finalfrontier/src/sgd.rs

```diff
@@ -1,11 +1,11 @@
 use ndarray::{Array1, ArrayView1, ArrayViewMut1};
 
+use crate::loss::log_logistic_loss;
+use crate::train_model::{NegativeSamples, TrainIterFrom, Trainer};
+use crate::vec_simd::scaled_add;
 use hogwild::Hogwild;
-use loss::log_logistic_loss;
-use train_model::{NegativeSamples, TrainIterFrom, Trainer};
-use vec_simd::scaled_add;
 
-use TrainModel;
+use crate::TrainModel;
 
```

Comment:
`crate::` `hogwild::` followed by a new block with a single `crate::` import, I guess we can clean up some of these imports? I'd suggest throwing the `TrainModel` import into the `train_model::{}` statement. (I'm probably the one responsible for these imports anyways).

Reply:
Indeed, I saw this in some other places as well. Let's do this in a separate PR. But probably after I've reviewed your PR, I should avoid making it too difficult for you to rebase things. (I made these changes on `master` and `release-0.5`, because finalfrontier didn't even build with Rust 1.32, because of using some 1.33 features.)